common/src/bitset.rs

```diff
@@ -28,56 +31,101 @@ impl IntoIterator for TinySet {
 }
 
 impl TinySet {
+    pub fn serialize<T: Write>(&self, writer: &mut T) -> io::Result<()> {
+        writer.write_all(self.0.to_le_bytes().as_ref())
+    }
+
+    #[inline]
+    pub fn deserialize(data: [u8; 8]) -> io::Result<Self> {
+        let val: u64 = u64::from_le_bytes(data);
+        Ok(TinySet(val))
+    }
+
     /// Returns an empty `TinySet`.
+    #[inline]
     pub fn empty() -> TinySet {
         TinySet(0u64)
     }
 
+    /// Returns a full `TinySet`.
+    #[inline]
+    pub fn full() -> TinySet {
+        TinySet::empty().complement()
+    }
+
     pub fn clear(&mut self) {
         self.0 = 0u64;
     }
 
+    #[inline]
     /// Returns the complement of the set in `[0, 64[`.
-    fn complement(self) -> TinySet {
+    pub fn complement(self) -> TinySet {
         TinySet(!self.0)
     }
 
-    /// Returns true iff the `TinySet` contains the element `el`.
-    pub fn contains(self, el: u32) -> bool {
-        !self.intersect(TinySet::singleton(el)).is_empty()
+    #[inline]
+    /// Returns true iff the `TinySet` bit is set at position `pos`.
+    pub fn contains(self, pos: u32) -> bool {
+        !self.intersect(TinySet::singleton(pos)).is_empty()
     }
 
-    /// Returns the number of elements in the TinySet.
-    pub fn len(self) -> u32 {
+    #[inline]
+    /// Returns the number of set bits in the TinySet.
+    pub fn num_set(self) -> u32 {
         self.0.count_ones()
     }
 
+    #[inline]
+    /// Returns the number of unset bits in the TinySet.
+    pub fn num_unset(self) -> u32 {
+        self.0.count_zeros()
+    }
+
+    #[inline]
     /// Returns the intersection of `self` and `other`
     pub fn intersect(self, other: TinySet) -> TinySet {
         TinySet(self.0 & other.0)
     }
 
-    /// Creates a new `TinySet` containing only one element
+    /// Creates a new `TinySet` with only one bit set at `pos`.
```

Comment:
Again... element was perfectly fine. It's a set. It has element.

I understand it is possibly  to sell it as a bitfield, but I  don't think it is better, and certainly not useful to rename the world partially for it.